ntbs-service-unit-tests/Services/NtbsSearchBuilderTest.cs

```diff
@@ -1,313 +1,513 @@
 using ntbs_service.Models.Entities;
 using ntbs_service.Services;
+using ntbs_service_unit_tests.DataAccess;
 using Xunit;
 
 namespace ntbs_service_unit_tests.Services
 {
     public class NtbsSearchBuilderTest
     {
-        readonly NtbsSearchBuilder builder;
+        private DbContextOptions<NtbsContext> ContextOptions { get; }
 
         public NtbsSearchBuilderTest()
         {
-            IQueryable<Notification> notifications = (new List<Notification> {
-                new Notification {
-                    NotificationId = 1,
-                    ETSID = "12",
-                    LTBRID = "222",
-                    LTBRPatientId = "22",
-                    NotificationDate = new DateTime(2000, 1, 1),
-                    PatientDetails = new PatientDetails {
-                        FamilyName = "Merry",
-                        GivenName = "Christmas",
-                        NhsNumber = "1234567890",
-                        SexId = 1,
-                        CountryId = 1,
-                        Dob = new DateTime(1990, 1, 1),
-                        PostcodeToLookup = "SW12RT"
-                    },
-                    HospitalDetails = new HospitalDetails {
-                        TBServiceCode = "Ashford hospital"
-                    }
-                },
-                new Notification {
-                    NotificationId = 2,
-                    ETSID = "13",
-                    LTBRID = "223",
-                    LTBRPatientId = "23",
-                    NotificationDate = new DateTime(2001, 1, 1),
-                    PatientDetails = new PatientDetails {
-                        FamilyName = "Merry",
-                        GivenName = "Goround",
-                        NhsNumber = "1234567891",
-                        SexId = 2,
-                        CountryId = 2,
-                        Dob = new DateTime(1991, 1, 1),
-                        PostcodeToLookup = "SW294FB"
-                    },
-                    HospitalDetails = new HospitalDetails {
-                        TBServiceCode = "Not Ashford"
-                    }
-                },
-            }).AsQueryable();
-
-            builder = new NtbsSearchBuilder(notifications);
+            ContextOptions = new DbContextOptionsBuilder<NtbsContext>()
+                .UseInMemoryDatabase(nameof(NtbsSearchBuilderTest))
+                .Options;
         }
 
         [Fact]
-        public void SearchById_ReturnsMatchOnNotificationId()
+        public async Task SearchById_ReturnsMatchOnNotificationId()
         {
-            var result = ((INtbsSearchBuilder)builder.FilterById("1")).GetResult().ToList();
-
-            Assert.Single(result);
-            Assert.Equal(1, result.FirstOrDefault().NotificationId);
+            using (var context = new NtbsContext(ContextOptions))
+            {
+                await context.Database.EnsureDeletedAsync();
+                await context.Database.EnsureCreatedAsync();

```

Comment:
I'm concerned that deleting and creating the database for each test will be quite slow. How long do these tests take to run?

Could we do all these tests on a single context instead? Or at least only do the delete/create once?

Reply:
See https://xunit.net/docs/shared-context (a ClassFixture may well be the right choice for this)